client/src/app/api/rest.ts

```diff
@@ -748,3 +750,37 @@ export const updateQuestionnaire = (
 // TODO:  of 204 - NoContext) ... the return type does not make sense.
 export const deleteQuestionnaire = (id: number): Promise<Questionnaire> =>
   axios.delete(`${QUESTIONNAIRES}/${id}`);
+
+export const getApplicationAnalysisReport = (
+  id: number,
+  type: MimeType
+): Promise<void> =>
+  axios
+    .get(`${APPLICATIONS}/${id}/analysis/report`, {
+      responseType: "blob",
+      headers: {
+        Accept: `application/x-${type}`,
+      },
+    })
+    .then((response) => {
+      if (response.status !== 200) {
+        throw new Error("Network response was not ok when downloading file.");
+      }
+      response.data;
+    })
+    .catch((error) =>
+      console.error("There was an error downloading the file:", error)
+    );
+
+export const getYamlQuestionnaire = (id: number): Promise<void> =>
```

Comment:
Changing the name to indicate that this is a download endpoint would be better I think. And removing yaml from the name of the functions would probably be best just incase we need to support other types.